File: 03- Subprogramas y recursividad/subprogramas/java/TablasMultiplicar.java
/**
 * Descripción: Solicita un número entero positivo y muestra su tabla de multiplicar del 1 al
 *      10. Si el número no es válido, vuelve a solicitarlo.
 * Autor: Santiago Nicolás De la mora Núñez
 * Fecha: 10/02/2026
 */

import javax.swing.JOptionPane;

public class TablasMultiplicar {
    public static void main(String[] args) {
        int valor = solicitarValor();
        JOptionPane.showMessageDialog(null, generarTablas2(valor), "Tabla", JOptionPane.PLAIN_MESSAGE);
    }

    public static int solicitarValor() {
        int numero = 0;
        boolean esValido = true;

        do {
            try {
                numero = Integer.parseInt(JOptionPane.showInputDialog(null, "Ingrese un número", "Número", JOptionPane.PLAIN_MESSAGE));

                if (numero > 0) {
                    esValido = true;
                } else {
                    JOptionPane.showMessageDialog(null, "No puede ingresar un número negativo o cero.", "Error", JOptionPane.ERROR_MESSAGE);
                    esValido = false;
                }
            } catch (NumberFormatException exception) {
                JOptionPane.showMessageDialog(null, "Debes introducir un número entero.", "Error", JOptionPane.ERROR_MESSAGE);
                esValido = false;
            }
        } while (!esValido);

        return numero;
    }

    // Este método resuelve la impresión de la tabla
    public static void generarTablas(int numero) {
        for (int i = 1; i <= 10; i++) {
            System.out.println(numero + " x " + i + " = " + (i * numero));
        }
    }

    public static String generarTablas2(int numero) {
        String tabla = "";

        for (int i = 1; i <= 10; i++) {
            tabla += numero + " x " + i + " = " + (i * numero) + "\n";
        }

        return tabla;
    }
}

File: 03- Subprogramas y recursividad/subprogramas/python/tablas_multiplicar.py
"""
Descripción: Solicita un número entero positivo y muestra su tabla de multiplicar del 1 al
    10. Si el número no es válido, vuelve a solicitarlo.
Autor: Santiago Nicolás De la mora Núñez
Fecha de creación: 10/04/2026
"""

def main():
    valor = solicitar_valor()
    print(generar_tablas2(valor))

def solicitar_valor():
    numero = 0
    es_valido = True

    while True:
        try:
            numero = int(input("Ingrese un número: "))

            if numero > 0:
                es_valido = True
            else:
                print("\nError. No puede ingresar un número negativo o 0.\n")
                es_valido = False
        except ValueError:
            print("\nError. Debes introducir un número entero.\n")
            es_valido = False

        if es_valido:
            break

    return numero

def generar_tablas(numero):
    i = 1
    while i <= 10:
        print(f"{numero} x {i} = {numero * i}")

        i += 1

def generar_tablas2(numero):
    tabla = ""

    i = 1
    while i <= 10:
        tabla += f"{numero} x {i} = {numero * i}\n"

        i += 1

    return tabla

# Ejecuta la aplicación
if __name__ == '__main__':
    main()